src/Units/UnitModelScheme.java

```diff
@@ -38,15 +41,22 @@ public final GameReadyModels getGameReadyModels()
   }
   protected abstract GameReadyModels buildGameReadyModels();
 
+  public void registerStateTrackers(GameInstance gi)
+  {
+    for( UnitModel um : gi.rules.unitModelScheme.getGameReadyModels().unitModels )
+      for( UnitModifier mod : um.getModifiers() )
+        mod.registerTrackers(gi);
+  }
```

Comment:
Dare I ask why a `UnitModelScheme` would need to register state trackers? Isn't a scheme immutable? Why would it need to know what's going on in the game? Maybe I just haven't gotten there yet, but it feels like we are breaking some assumptions here.
One of which might be: Vanilla units in a `UnitModelScheme` not having modifiers.

Also, why `gi.rules.unitModelScheme.getGameReadyModels()` instead of just `getGameReadyModels()`?
For that matter, maybe this should be moved to `GameInstance.init()` or something.

Reply:
Sure. If a particular unit type needs to have extra information about the gamestate in order to handle its functionality, then it needs a `StateTracker` and can't depend on a CO or something to initialize it. The UMS knows about the unit and can initialize the tracker.
Schemes are immutable, yes. This isn't really for the *scheme* so much as the units in it. That's probably not obvious when there's no implementation, sorry.
> Vanilla units in a `UnitModelScheme` not having modifiers.

So uh. I have been planning for a while to introduce something like this. That's why the `UnitModel` supports its own list of modifiers even though COs aren't allowed to mess with it.
An example I've been thinking on is the lose-range-with-movement effect on the Bship. It *could* be a property on the weapon that every piece of code that touches it has to enforce, or it could be a `UnitModifier` and be handled the same way CO effects are.

Uhhhh, I originally had this code in a static function in another class, I think? Copy/paste error from there.
I'm not super opposed to pulling it up to `GameInstance.init()`, but I think the way it is now is also useful.
For example, all Kaiju types use the same tracker for tracking ability cooldowns - who registers that tracker? You could register it in each Kaiju unit definition, or you could just register it in the UMS.